src/main/java/dan200/computercraft/api/ComputerCraftAPI.java

```diff
@@ -221,6 +222,21 @@ public static void registerAPIFactory( @Nonnull ILuaAPIFactory factory )
         getInstance().registerAPIFactory( factory );
     }
 
+    /**
+     * Registers a detail provider to provide additional details for blocks, fluids and items when inspected by methods
+     * such as {@code turtle.getItemDetail()} or {@code turtle.inspect()}.
+     *
+     * @param provider The detail provider to register.
+     * @param <T> The type of object that this provider can provide details for.
+     * @see dan200.computercraft.api.detail.IBlockDetailProvider
+     * @see dan200.computercraft.api.detail.IFluidDetailProvider
+     * @see dan200.computercraft.api.detail.IItemDetailProvider
+     */
+    public static <T> void registerDetailProvider( @Nonnull IDetailProvider<T> provider )
```

Comment:
Rather than having three sub-interfaces, I wonder if it's just cleaner to only have `IDetailProvider<T>` and either have three kinds of registration method, or just a `registerDetailProvider(Class<T>, IDetailProvider<T>)`. This has the added advantage that you can register with a lambda without having to specify a class.